site/en/docs/extensions/mv3/match_patterns/index.md

```diff
@@ -1,50 +1,156 @@
+- **host**: A hostname (`www.example.com`), a `*` before the hostname to match subdomains (`*.example.com`), or just a wildcard `*`.
 
-Here are some examples of _invalid_ pattern matches:
+- **path**: Must start with `/` and be present. If alone it will always treated as `/*`. For example `/*`, `/foo*`, or `/foo/bar`. Each '`*`' matches 0 or more characters.
 
-<table class="fixed-table width-full"><tbody><tr><th style="margin-left:0; padding-left:0">Bad pattern</th><th style="margin-left:0; padding-left:0">Why it's bad</th></tr><tr><td><code>https://www.google.com</code></td><td>No <em>path</em></td></tr><tr><td><code>https://*foo/bar</code></td><td>'*' in the <em>host</em> can be followed only by a '.' or '/'</td></tr><tr><td><code>https://foo.*.bar/baz&nbsp;</code></td><td>If '*' is in the <em>host</em>, it must be the first character</td></tr><tr><td><code>http:/bar</code></td><td>Missing <em>scheme</em> separator ("/" should be "//")</td></tr><tr><td><code>foo://*</code></td><td>Invalid <em>scheme</em></td></tr></tbody></table>
+## Special cases {: #special }
 
-Some schemes are not supported in all contexts.
+`<all_urls>`
+: It matches any URL that starts with a permitted scheme. This is considered a broad host permission.
 
-[1]: /docs/extensions/mv3/declare_permissions#host-permissions
-[2]: /docs/extensions/mv3/content_scripts
+`file:///`
+: Allows your extension to run on local files. It has no host and requires the user to manually [enable access][permissions].
+
+Port URLs
+: Use `http://localhost/*` to match any localhost ports during development. For IP addresses, include the specific address plus a wildcard in the path. For example: `http://127.0.0.1/*`
```

Comment:
This is not quite clear to me. Would be good to give an example that includes. a port definition.

Reply:
@sebastianbenz Maybe port is not the term we should use here? I was going by the terminology used in [this issue](https://github.com/GoogleChrome/developer.chrome.com/issues/2250). 🤔

Should we change the title to **Localhost URLs and IP Addresses**?

Reply:
What confused me is that ports are mixed with localhost / ipaddresses, but both concepts are independent, e.g. `https://example.com:8081` is valid. Are the match patterns specific to localhost? 

What I'm wondering is, how can I match the following URLs:

```
http://localhost:8080
http://localhost:8080/my-path
http://127.0.0.1:8080
http://127.0.0.1:8080/my-path
https://example.com:8081
https://example.com:my-path
```